homeassistant/components/ecobee/climate.py

```diff
@@ -114,20 +110,27 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Ecobee Thermostat Platform."""
-    if discovery_info is None:
-        return
-    data = ecobee.NETWORK
-    hold_temp = discovery_info["hold_temp"]
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Old way of setting up ecobee thermostat."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the ecobee thermostat."""
+    data = hass.data[DOMAIN]
+    hold_temp = config_entry.options[CONF_HOLD_TEMP]
+    devices = list()
+
     _LOGGER.info(
         "Loading ecobee thermostat component with hold_temp set to %s", hold_temp
     )
-    devices = [
-        Thermostat(data, index, hold_temp)
-        for index in range(len(data.ecobee.thermostats))
-    ]
-    add_entities(devices)
+
+    for index in range(len(data.ecobee.thermostats)):
+        dev = Thermostat(data, index, hold_temp)
+        config_entry.add_update_listener(dev.option_update_listener)
```

Comment:
Please subscribe the listener in entity coroutine method `async_added_to_hass`.

We should unsubscribe the listener in `async_will_remove_from_hass`. `add_update_listener` returns a function to unsubscribe the listener.

Reply:
OK so I was just thinking about that - if I need to subscribe in async_added_to_hass, that means I need to pass the config entry into the instance... if I'm doing that, can't we just save some trouble and do something like:
```python
@property
def hold_temp(self):
    return self._config_entry.options[CONF_HOLD_TEMP]
```
In which case, the hold_temp property will always return the updated option value.

Reply:
Yes, that should work.